### DISCUSSION TITLE
Page index does not count up [Dart/Flutter]

### DISCUSSION THREAD

**Discussion by RahmiTufanoglu** [pocketbase/pocketbase#1016] [Q&A] [answered]
I noticed a strange behaviour with the pagination. 

I have in total 13 data and fetch 10 per page.
Page 1: 10
Page 2: 3

But for some reason after counting up the page index to 3 the response is still on page 2 with the data from page 2.
If I set the page index to 99 the response still shows me the data from page 2.

Here is the code snippet:

```
await client.admins.authViaEmail('admin@mail.com', 'Test.12345');

final result = await client.collections.getList(
      'users',
      page: page,
      perPage: 10,
      sort: '-created',
);
```

**Comment by ganigeorgiev** [comment, marked as answer]
I think this is expected. As far as I understand from the example - you have only 13 items total which is split in 2 pages (10 and 3). Requesting any page higher than the max existing will always return the max available page, in your case - that is the second one.

**Reply by ganigeorgiev** [reply]
We also have tests for that (both unit and integration). Here is the unit one - https://github.com/pocketbase/pocketbase/blob/master/tools/search/provider_test.go#L309-L322

**Comment by RahmiTufanoglu** [comment]
@ganigeorgiev And how should I properly check if I reach the end of the list?

**Reply by ganigeorgiev** [reply]
The list response should have `totalPages` and you can compare it to the current `page`. All fields of the `ResultList` DTO could be found in https://github.com/pocketbase/dart-sdk/blob/master/lib/src/dtos/result_list.dart
